dataactvalidator/scripts/loadTas.py

```diff
@@ -74,44 +109,97 @@ def update_tas_lookups(csv_path):
     new_data = data[data['existing_id'].isnull()]
     del new_data['existing_id']
 
-    # instead of using the pandas to_sql dataframe method like some of the
-    # other domain load processes, iterate through the dataframe rows so we
-    # can load using the orm model (note: toyed with the SQLAlchemy bulk load
-    # options but ultimately decided not to go outside the unit of work for
-    # the sake of a performance gain)
-    for _, row in old_data.iterrows():
-        sess.query(TASLookup).filter_by(account_num=row['account_num']).update(row, synchronize_session=False)
-
-    for _, row in new_data.iterrows():
-        sess.add(TASLookup(**row))
+    if update_missing:
+        # find which incoming records can fill in the empty records
+        relevant_old_data = old_data[old_data['account_num'].isin(update_missing)]
+        # Fill them in. If budget_function_code is empty, the following columns have also been empty.
+        fill_in_cols = ['account_title', 'budget_bureau_code', 'budget_bureau_name', 'budget_function_code',
+                        'budget_function_title', 'budget_subfunction_code', 'budget_subfunction_title',
+                        'reporting_agency_aid', 'reporting_agency_name']
+        for _, row in relevant_old_data.iterrows():
+            fill_in_updates = {fill_in_col: row[fill_in_col] for fill_in_col in fill_in_cols}
+            fill_in_updates['updated_at'] = datetime.now(timezone.utc)
+            sess.query(TASLookup).filter_by(account_num=row['account_num']).update(synchronize_session=False,
+                                                                                   values=fill_in_updates)
+        logger.info('%s records filled in', len(relevant_old_data.index))
+    else:
+        # instead of using the pandas to_sql dataframe method like some of the other domain load processes, iterate
+        # through the dataframe rows so we can load using the orm model (note: toyed with the SQLAlchemy bulk load
+        # options but ultimately decided not to go outside the unit of work for the sake of a performance gain)
+        for _, row in old_data.iterrows():
+            sess.query(TASLookup).filter_by(account_num=row['account_num']).update(row, synchronize_session=False)
+
+        for _, row in new_data.iterrows():
+            sess.add(TASLookup(**row))
+        logger.info('%s records in CSV, %s existing', len(data.index), sum(data['existing_id'].notnull()))
 
     sess.commit()
-    logger.info('%s records in CSV, %s existing', len(data.index), sum(data['existing_id'].notnull()))
 
 
-def load_tas(tas_file=None):
-    """Load TAS file into broker database. """
-    # read TAS file to dataframe, to make sure all is well
-    # with the file before firing up a db transaction
+def load_tas(tas_file=None, backfill_historic=False):
```

Comment:
At no point do we use the `tas_file` argument, there isn't even code for it to be used. In fact, there isn't even an `else` to handle what to do if we have a `tas_file`. Do we really need to keep it or any logic pertaining to it?

Reply:
I was keeping it in on the off chance we provide a local argument. It can be removed for now.